client/modules/sheaf/src/render-handler.ts

```diff
@@ -0,0 +1,127 @@
+import type { Text } from "@codemirror/state"
+import * as FTML from "ftml-wasm-worker"
+import { Memoize } from "typescript-memoize"
+import { toFragment } from "wj-util"
+
+/**
+ * Heavily memoized FTML-render-emit handler.
+ * Designed to invoke the FTML renderer as little as possible.
+ */
+export class RenderHandler {
+  private declare fragmentNode?: DocumentFragment
+
+  /** @param doc - The CodeMirror `state.doc` to render with. */
+  constructor(public doc?: Text) {}
```

Comment:
```suggestion
  constructor(private doc?: Text) {}
```
Changing it would do nothing, assuming the wanted method has already been called at least once - might be better to prevent potentially inconsistent behaviour? Or will mutability be important here?

Reply:
You're correct about mutability but rather than making it private it should be made `readonly`. I'll add that in another PR because I'd rather not deal with merge conflicts